iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/selector/estimator/AbstractCompactionEstimator.java

```diff
@@ -140,8 +142,12 @@ protected int calculatingMaxOverlapFileNumInSubCompactionTask(List<TsFileResourc
       long maxEndTimeOfCurrentDevice = Long.MIN_VALUE;
       int overlapFileNumOfCurrentDevice = 0;
       for (ArrayDeviceTimeIndex resource : resourcesContainsCurrentDevice) {
-        long deviceStartTimeInCurrentFile = resource.getStartTime(device);
-        long deviceEndTimeInCurrentFile = resource.getEndTime(device);
+        if (resource.definitelyNotContains(device)) {
```

Comment:
This check should have been done in line135

Reply:
Removed